Palesun daylight skin

michelealbano opened the pull request:
I developed a python script that visits the folders containing a skin, and re-creates in the current folder the same folder structure, copies locally the png / psd files, and for everything else it copies the files but inverts the colors.

This PR could solve issue https://github.com/mixxxdj/mixxx/issues/9217

Commit:
Added a tool (intertcolor.py) to invert all the colors in a file, to be
used with skins.

Commit:
Added a skin (PaleSun) that is the inverted color version of PaleMoon.
Its colors are daylight-friendly.

Commit:
Added the generated color scheme "PaleSun" for the LateNight skin

Commit:
Added my name to skin.xml

Commit:
applying the coding standard to "invertcolor.py" file

Commit:
Another try at satisfying the code standards

Commit:
changing newlines for the coding standard

JoergAtGithub commented:
Why did you opened a new PR, instead of fixing the old one? This is difficult to track for reviewers.

Commit:
pre-commit checks used

JoergAtGithub commented:
If pre-commit prints that it modified files, did you commit the original or the modified files? Depending on you Git client, you might need to stage them manually again.

michelealbano commented:
I didn't know you can change the branch you commit too without
creating a new PR.


On Sun, Jul 16, 2023 at 10:02 PM JoergAtGithub ***@***.***> wrote:
>
> Why did you opened a new PR, instead of fixing the old one? This is difficult to track for reviewers.
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the thread.Message ID: ***@***.***>

Commit:
more corrections for the coding style

daschuer commented:
Nice. This is a good start. Thank you.

Did you do some tests under direct sunlight conditions?
I think we should do this and optimize the skin a bit. 

Can we make the library text color black? 

Do you also have interest to restyle the Shade library as well? The Skin itself is OK with sunlight, but the library is only barley readable.

michelealbano commented:
I used this skin under sunlight in 3 events and it went well, I was able to
see everything I needed. I have not tried it with a CDJ yet, I have a DDJ
FLX4.

I will take a look at the library text colors in a few weeks, after Summer
vacation.

Regarding the Shade skin, I tried to work on it but the color inversion
made it ugly and not very usable.



On Mon, Jul 17, 2023 at 12:33 AM Daniel Schürmann ***@***.***>
wrote:

> Nice. This is a good start. Thank you.
>
> Did you do some tests under direct sunlight conditions?
> I think we should do this and optimize the skin a bit.
>
> Can we make the library text color black?
>
> Do you also have interest to restyle the Shade library as well? The Skin
> itself is OK with sunlight, but the library is only barley readable.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mixxxdj/mixxx/pull/11748#issuecomment-1637205926>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC66LHEP4IJEEPMJWRM6CNLXQRT4XANCNFSM6AAAAAA2MEHCRM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

daschuer commented:
Thank you. 

For Shade I meant to just make the library brighter.

Commit:
copyright etc were shortened, as suggested by daschuer

michelealbano commented on tools/invertcolor.py:
```diff
@@ -0,0 +1,122 @@
+__version__ = "0.91"
+__maintainer__ = "Michele Albano"
+__email__ = "michele.albano@gmail.com"
+__status__ = "Prototype"
```
ok

ronso0 commented on res/skins/LateNight/style_palesun.qss:
```diff
@@ -0,0 +1,3119 @@
+  background-color: #f7f7f7;
+}
+
+#213k1, #213kCompact1, #213kMini1,
```
There are some strange replacements which basically eliminate a lot of styles:
`#Deck` > `#213k`
`#Beat` > `#415t`

michelealbano commented:
Thank you for your comment. Palesun is intended to be a skin/style for
people DJing under a not-so-pale sun, and I privileged function over form.
I am open to suggestions, but please try out both Palesun and your modified
version under the sun, to check that protecting the style does not impair
vision.



On Thu, 27 Jul 2023, 14.31 ronso0, ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In res/skins/LateNight/style_palesun.qss
> <https://github.com/mixxxdj/mixxx/pull/11748#discussion_r1276211850>:
>
> > +* colors & icons
> +  * font colors
> +  * button styles
> +  * button icons
> +
> +* library
> +* menu & special widget styles
> +
> +
> +/**********************************************************************
> +************** container background colors & borders *****************/
> +#SkinContainer {
> +  background-color: #f7f7f7;
> +}
> +
> +#213k1, #213kCompact1, #213kMini1,
>
> There are some strange replacements which basically eliminate a lot of
> styles:
> #Deck > #213k
> #Beat > #415t
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mixxxdj/mixxx/pull/11748#pullrequestreview-1549738351>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC66LHDV2LSLT4GPEGG7CUTXSJNQ3ANCNFSM6AAAAAA2MEHCRM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

michelealbano commented:
I have reduced it.

On Mon, 17 Jul 2023, 07.00 Daniel Schürmann, ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In tools/invertcolor.py
> <https://github.com/mixxxdj/mixxx/pull/11748#discussion_r1264861285>:
>
> > +# GNU General Public License Usage
> +# This file may be used under the terms of the GNU
> +# General Public License version 3 as published by the Free Software
> +# Foundation. Please review the following information to
> +# ensure the GNU General Public License version 3 requirements
> +# will be met: https://www.gnu.org/licenses/gpl-3.0.html.
> +
> +__author__ = "Michele Albano"
> +__copyright__ = "Copyright 2023"
> +__credits__ = ["Michele Albano"]
> +__license__ = "GPL"
> +__version__ = "0.91"
> +__maintainer__ = "Michele Albano"
> +__email__ = ***@***.***"
> +__status__ = "Prototype"
>
> Is it possible to remove or reduce this? We normally track the copyright
> by the git commits and have a single License file, which is still at GPL 2.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mixxxdj/mixxx/pull/11748#pullrequestreview-1531961137>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC66LHA2B7KBSH6RQQLQZTLXQTBHRANCNFSM6AAAAAA2MEHCRM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

michelealbano commented:
Ok I understand now what you mean, it's hilarious and I will correct that

On Thu, 27 Jul 2023, 14.31 ronso0, ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In res/skins/LateNight/style_palesun.qss
> <https://github.com/mixxxdj/mixxx/pull/11748#discussion_r1276211850>:
>
> > +* colors & icons
> +  * font colors
> +  * button styles
> +  * button icons
> +
> +* library
> +* menu & special widget styles
> +
> +
> +/**********************************************************************
> +************** container background colors & borders *****************/
> +#SkinContainer {
> +  background-color: #f7f7f7;
> +}
> +
> +#213k1, #213kCompact1, #213kMini1,
>
> There are some strange replacements which basically eliminate a lot of
> styles:
> #Deck > #213k
> #Beat > #415t
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mixxxdj/mixxx/pull/11748#pullrequestreview-1549738351>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC66LHDV2LSLT4GPEGG7CUTXSJNQ3ANCNFSM6AAAAAA2MEHCRM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

ronso0 commented:
The script is definitly helpful for skteches, even if it has some flaws (see string replacement above).
However, plain color conversion also inverts drop shadows of containers, buttons and knobs which I don't like tbh. Fixing those manually is a lot of work, e.g. flipping the button border SVGs. It would be nice if that could be scripted, but definitly out of scope IMO. We may also make the scheme flat, i.e. ditch all 3D effects and focus more on high contrast.
For example this:
![image](https://github.com/mixxxdj/mixxx/assets/5934199/cf8bb9a3-0c44-4804-8767-cbdfab143480)


That said, if we want to add a daylight scheme I'd rather go back to the cutting table and design something new.

Commit:
corrected PaleSun style

the tool for generating the PaleSun style for LateNight skin (invertcolor.py)
was reworked, now it can correctly invert the colors

michelealbano commented:
> Ok I understand now what you mean, it's hilarious and I will correct that
> […](#)
> On Thu, 27 Jul 2023, 14.31 ronso0, ***@***.***> wrote: ***@***.**** commented on this pull request. ------------------------------ In res/skins/LateNight/style_palesun.qss <[#11748 (comment)](https://github.com/mixxxdj/mixxx/pull/11748#discussion_r1276211850)>: > +* colors & icons + * font colors + * button styles + * button icons + +* library +* menu & special widget styles + + +/********************************************************************** +************** container background colors & borders *****************/ +#SkinContainer { + background-color: #f7f7f7; +} + +#213k1, #213kCompact1, #213kMini1, There are some strange replacements which basically eliminate a lot of styles: #Deck > #213k #Beat > #415t — Reply to this email directly, view it on GitHub <[#11748 (review)](https://github.com/mixxxdj/mixxx/pull/11748#pullrequestreview-1549738351)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AC66LHDV2LSLT4GPEGG7CUTXSJNQ3ANCNFSM6AAAAAA2MEHCRM> . You are receiving this because you authored the thread.Message ID: ***@***.***>

I have corrected the issue raised by @ronso0 and I have already tried it. Please check it out

Commit:
style corrected

luzpaz commented:
Is it possible to see more screenshots ?

luzpaz commented:
bump

Holzhaus commented:
@luzpaz you can just download the skin and put it in your skin directory to test it yourself.

michelealbano commented:
> @luzpaz you can just download the skin and put it in your skin directory to test it yourself.

I tried to create some screenshots, but for that I updated my mixxx development branch. I am not able to compile it anymore. I use windows, and the cmake of visual studio is outdated with respect to what is required. Yes, I have just updated Visual Studio 2019, and it did not solve the issue.

I will add screenshots as soon as I find a cmake that can do the work.

daschuer commented:
Please post the build output. 
We are working hard to make everything work out of the box. Maybe something is missing for your case.

acolombier commented:
Hey @michelealbano, are you still looking at completing this? 

I think adding a light variation of PaleMoon sounds like a great usecase, for outdoor performances under a bright sun, even if it is sub-optimal. Since we are aiming to eventually move to QML, it's probably best to make an automation translation, rather than spending sometime building from the ground up 

I'll make that a draft PR for now, but feel free to mark it ready if it should be reconsidered for merging.

ronso0 commented:
> I think adding a light variation of PaleMoon sounds like a great usecase, for outdoor performances under a bright sun

Yeah, sure!
> even if it is sub-optimal

Sorry, I still disagree here. Even if we ignore the asthetics, there are still too many sections/controls that are barely readable in bright light (Fx super knob arc, fx mixmode icons, slider scales, fx assign button text) that I don't feel comfortable with publishing this theme.
Though the flaws can be fixed of course.

acolombier commented:
Fair enough - @michelealbano do you have interest in looking into the readability issues?

ronso0 commented:
FWIW I just dug out my HighNoon draft again and continued to tweak it.
Focusing on high contrast here. Nowhere near finished but you get the idea!
![image](https://github.com/user-attachments/assets/14437e25-c4a5-4779-be43-3d1be48df392)


Simplified knobs are next, then samplers and effects, then the library.
Will open a WIP PR in the next week(s), seems feasible to get it done for 2.6

acolombier commented:
This looks promising! I will close this PR for now. Thanks for initiating this innovation @michelealbano and thanks for picking this up @ronso0 !

michelealbano commented:
ok no prob. I am still using my own "inverted colors" mixxx in the events I hold outdoors, but I will take a look at HighNoon. Where can I find it?

ronso0 commented:
Yup, it's here https://github.com/ronso0/mixxx/tree/highnoon
But the library for example is not usable atm.

ronso0 commented:
I revived it in #16404
